release_tools/changelog.py

```diff
@@ -0,0 +1,329 @@
+            'HOME': os.getenv('HOME', '')
+        }
+
+    @property
+    def root_path(self):
+        path_ = self._get_submodule_root_path()
+
+        if not path_:
+            basepath = self._get_root_path()
+
+        return basepath
+
+    def _get_root_path(self):
+        cmd = ['git', 'rev-parse', '--show-toplevel']
+        root_path = self._exec(cmd, env=self.gitenv).strip('\n')
+        return root_path
+
+    def _get_submodule_root_path(self):
+        cmd = ['git', 'rev-parse', '--show-superproject-working-tree']
+        root_path = self._exec(cmd, env=self.gitenv).strip('\n')
+        return root_path
+
+    @staticmethod
+    def _exec(cmd, cwd=None, env=None):
+        proc = subprocess.Popen(cmd, stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE,
+                                cwd=cwd, env=env)
+        (outs, errs) = proc.communicate()
+        return outs.decode('utf-8', errors='surrogateescape')
+
+
+@enum.unique
+class CategoryChange(enum.Enum):
+    """Possible category types."""
+
+    def __new__(cls, value, category, title):
+        obj = object.__new__(cls)
+        obj._value_ = value
+        obj.category = category
+        obj.title = title
+        return obj
+
+    ADDED = (1, 'added', 'New feature')
+    FIXED = (2, 'fixed', 'Bug fix')
+    CHANGED = (3, 'changed', 'Feature change')
+    DEPRECATED = (4, 'deprecated', 'New deprecation')
+    REMOVED = (5, 'removed', 'Feature removal')
+    SECURITY = (6, 'security', 'Security fix')
+    PERFORMANCE = (7, 'performance', 'Performance improvement')
+    OTHER = (8, 'other', 'Other')
+
+    @classmethod
+    def values(cls):
+        return [member.category for member in cls]
+
+
+class ChangelogEntry:
+    """Class to store changelog entries data."""
+
+    def __init__(self, title, category, author, pr=None, notes=None):
+        self.title = title
+        self.category = category
+        self.author = author
+        self.pr = pr
+        self.notes = notes
+
+    def to_dict(self):
+        return {
+            'title': self.title,
+            'category': self.category,
+            'author': self.author,
+            'pull_request': self.pr,
+            'notes': self.notes
+        }
+
+
+def title_prompt():
+    prompt_msg = ">> Please specify the title of your change"
+    return prompt_msg
+
+
+def category_prompt():
+    prompt_msg = ">> Please specify the category of your change\n\n"
+    prompt_msg += "\n".join(["{}. {}".format(cat.value, cat.title) for cat in CategoryChange])
+    prompt_msg += "\n\n"
+    return prompt_msg
+
+
+def validate_title(ctx, param, value):
+    """Check title option values."""
+
+    value = value.strip("\n\r ")
+
+    if not value:
+        raise click.BadParameter("title cannot be empty")
+    return value
+
+
+def validate_category(ctx, param, value):
+    """Check category option values.
+
+    Valid values for a category are integer indexes
```

Comment:
Maybe ctx, param and value and their description could appear in the docstring.

Reply:
Those are parameters that are not used in the function and are related to how `click` library works. I prefer to keep the docstring as simple as possible.